icu4c/source/i18n/chnsecal.cpp

```diff
@@ -1184,6 +1191,27 @@ ChineseCalendar::Setting ChineseCalendar::getSetting(UErrorCode&) const {
   };
 }
 
+int32_t
+ChineseCalendar::getActualMaximum(UCalendarDateFields field, UErrorCode& status) const
+{
+    if (U_FAILURE(status)) {
+       return 0;
+    }
+    if (field == UCAL_DATE) {
+        LocalPointer<ChineseCalendar> cal(clone(), status);
+        if(U_FAILURE(status)) {
+            return 0;
+        }
+        cal->setLenient(true);
+        cal->prepareGetActual(field,false,status);
+        int32_t year = cal->get(UCAL_EXTENDED_YEAR, status);
+        int32_t month = cal->get(UCAL_MONTH, status);
+        bool leap = cal->get(UCAL_IS_LEAP_MONTH, status) != 0;
+        return handleGetMonthLengthWithLeap(year, month, leap, status);
```

Comment:
I'm guessing this is a partial answer to my question is here, but if you just left `handleGetMonthLength()` as one function and updated it to handle the leap year, couldn't you just call it here (it'd then be getting the `UCAL_IS_LEAP_MONTH` field instead of getting it here).

Reply:
The problem is one usage need to use get (to force the calculation of all fields) and the other need to use internalGet (to avoid infinity recursion) and I cannot change the function signature (because it is defined in the base class as a protected method for all subclass of Calendar). The tricky part of this fix is it is very easy to get into infinity recursion

Reply:
Okay.  Thanks for the explanation.